INSTALL.md

```diff
@@ -194,7 +188,96 @@ Run [NPM](https://www.npmjs.com/) to install all the required Node modules:
 npm install
 ```
 
-#### Setup Database
+## Manual setup on Fedora (29)
+
+### Prerequisites
+
+#### Ruby version
+
+The project supports **Ruby 2.3.x**.
+
+Verify you have a proper version by running on your terminal:
+```bash
+ruby -v
+```
+
+> You can use [rbenv](https://github.com/rbenv/rbenv) or [rvm](https://rvm.io/) to install ruby.
+
+#### Node version
+
+The project supports **Version: 8.X.X**.
+
+Install Node version 8 and ensure that the node is properly configured in `PATH` environment variable.
+
+#### Dependencies
+
+```shell
+sudo yum install patch autoconf automake bison libffi-devel libtool libyaml-devel readline-devel sqlite-devel zlib-devel openssl-devel ImageMagick ImageMagick-devel mysql-devel postgresql-devel chromedriver memcached
+
+sudo systemctl restart memcached
+```
+
+#### Database (Postgres / MySQL / Oracle)
+
+Postgres, MySQL or Orcacle has to be running for the application to work. The easiest way to do it is in a [Docker](https://www.docker.com/) container by simply running:
+
+```shell
+docker run -d -p 5433:5432 -e POSTGRES_USER=postgres -e POSTGRES_DB=3scale_system_development --name postgres10 circleci/postgres:10.5-alpine
+```
+
+Alternatively you can run Postgres directly on your machine by following [this article](https://developer.fedoraproject.org/tech/database/postgresql/about.html).
+
+#### Redis
+
+[Redis](https://redis.io) has to be running for the application to work. The easiest way to do it is in a [Docker](https://www.docker.com/) container by simply running:
+
+```shell
+docker run -d -p 6379:6379 redis
+```
+
+Alternatively you can run Redis directly on your machine by using `yum`:
+
+```shell
+sudo yum install redis
+
+sudo systemctl restart redis
+```
+
+### Setup
+
+#### Eventmachine
+
+Eventmachine has to be installed with `--with-cppflags=-I/usr/include/openssl/`. Simply run:
+
+```shell
+bundle config build.eventmachine --with-cppflags=-I/usr/include/openssl/
+```
+
+#### Config files
+
+Copy example config files from the examples folder:
+
+```shell
+cp config/examples/* config/
+```
+
+#### Bundle
+
+Run [Bundler](https://bundler.io/) to install all required Ruby gems:
+
+```shell
+bundle install
+```
+
+#### NPM
+
+Run [NPM](https://www.npmjs.com/) to install all the required Node modules:
+
+```bash
+npm install
+```
+
+## Setup Database
```

Comment:
For the Provider  account, no random password gets generated. 

I executed the following command:
USER_PASSWORD=p MASTER_PASSWORD=p bin/rake db:drop db:setup
>>>> Everything worked fine on the terminal but on the login portal, password 'p' isn't accepted as a valid password. Both the USER_PASSWORD and MASTER_PASSWORD needs to be updated with a password of at least 6 characters, else login fails.
I think it's worth mentioning here.

Reply:
This is already fixed 😄 https://github.com/3scale/porta/pull/1163

Reply:
This is fixed now in master by https://github.com/3scale/porta/pull/1163